### PR TITLE
Rely only on explicit status

### PR DESCRIPTION
**Pull request by BlueSquare1**

### CODE REVIEW

**Review by elyousfi5 — APPROVED** [review]
Can you use explicit checks please, otherwise lgtm

File: agent/metasploit_agent.py [hunk 1]
```diff
@@ -92,52 +84,34 @@ def process(self, message: m.Message) -> None:
...
-            if isinstance(results, dict) and results.get("code") == "vulnerable":
-                technical_detail += f'Message: \n```{results["message"]}```'
-            else:
-                console_output = client.consoles.console(cid).run_module_with_output(
-                    module_instance
+            if isinstance(results, dict) and results.get("code") in [
+                "vulnerable",
+                "appears",
+            ]:
+                target = (
```
Comment (benyissa) [inline agent/metasploit_agent.py:104]: You can use the target class instead. I already added one in utils in my PR.

Comment (BlueSquare1) [reply agent/metasploit_agent.py:104]: No need as we only need a string of the target to show to the user

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/utils.py [hunk 2]
```diff
@@ -101,27 +101,42 @@ def _get_target_from_url(message: m.Message) -> Target | None:
...
+        port = _get_port(message, scheme)
+        try:
+            mask = int(message.data.get("mask", None))
+        except ValueError as exc:
+            raise ValueError("Invalid network mask provided") from exc
+        if mask is None:
+            hosts = ipaddress.ip_network(host)
+        else:
+            mask = int(mask)
+            if message.data.get("version") == 4 and mask < 24:
```
Comment (benyissa) [inline agent/utils.py:124]: I would put 24 and 120 as global constants.

File: agent/utils.py [hunk 3]
```diff
@@ -101,27 +101,42 @@ def _get_target_from_url(message: m.Message) -> Target | None:
...
+def prepare_targets(message: m.Message) -> list[Target]:
     """Prepare targets based on type, if a domain name is provided, port and protocol are collected
     from the config."""
     if (host := message.data.get("host")) is not None:
-        port = _get_port(message)
-        return host, port
+        scheme = _get_scheme(message)
+        port = _get_port(message, scheme)
+        try:
+            mask = int(message.data.get("mask", None))
```
Comment (benyissa) [inline agent/utils.py:117]: if mask is None this will lead to TypeError .
can you explain what does this code try to check please.

Comment (BlueSquare1) [reply agent/utils.py]: The code basically limits the CIDR range that the agent can scan to 255 hosts.

File: agent/utils.py [hunk 4]
```diff
@@ -101,27 +103,38 @@ def _get_target_from_url(message: m.Message) -> Target | None:
...
+        scheme = _get_scheme(message)
+        port = _get_port(message, scheme)
+        mask = message.data.get("mask", None)
+        if mask is None:
+            hosts = ipaddress.ip_network(host)
+        else:
+            if message.data.get("version") == 4 and int(mask) < IPv4_CIDR_LIMIT:
+                raise ValueError("Subnet mask below 24 is not supported.")
+            if message.data.get("version") == 6 and int(mask) < IPv6_CIDR_LIMIT:
+                raise ValueError("Subnet mask below 120 is not supported")
```
Comment (elyousfi5) [inline agent/utils.py:125]:
```suggestion
                raise ValueError(f"Subnet mask below {IPv6_CIDR_LIMIT} is not supported")
```

File: agent/utils.py [hunk 5]
```diff
@@ -101,27 +103,38 @@ def _get_target_from_url(message: m.Message) -> Target | None:
...
-        port = _get_port(message)
-        return host, port
+        scheme = _get_scheme(message)
+        port = _get_port(message, scheme)
+        mask = message.data.get("mask", None)
+        if mask is None:
+            hosts = ipaddress.ip_network(host)
+        else:
+            if message.data.get("version") == 4 and int(mask) < IPv4_CIDR_LIMIT:
+                raise ValueError("Subnet mask below 24 is not supported.")
```
Comment (elyousfi5) [inline agent/utils.py:123]:
```suggestion
                raise ValueError(f"Subnet mask below {IPv4_CIDR_LIMIT} is not supported")
```

Comment (BlueSquare1) [reply agent/utils.py]: Good catch!

**Review by amine3 — DISMISSED** [review]

File: agent/metasploit_agent.py [hunk 6]
```diff
@@ -89,62 +81,43 @@ def process(self, message: m.Message) -> None:
...
-
-            if isinstance(results, dict) and results.get("code") == "vulnerable":
-                technical_detail += f'Message: \n```{results["message"]}```'
-            else:
-                console_output = client.consoles.console(cid).run_module_with_output(
-                    module_instance
-                )
+            targets = utils.prepare_targets(message)
+            for target in targets:
+                rhost = target.host
```
[the remark this answers is not in the document]
Comment (BlueSquare1) [reply agent/metasploit_agent.py:86]: basically host and port, `rhost` and `rport` is just the naming scheme used in metasploit

**Review by 3asm — APPROVED** [review]

**Pull request merged**